### PR TITLE
[WIP] Setup of environment of publishing

### PR DESCRIPTION
**Pull request by Chun-MingChen**
#157

Hi @AllenFang, the following is the workflow for publishing, which takes refer to [storybook#release-guide](https://github.com/storybooks/storybook/blob/master/CONTRIBUTING.md#release-guide). Any suggestion and advices were welcomed :)

* Generate a `CHANGELOG`  and verify the release by hand. [WIP]
* Push the changelog to master or the release branch
* Clean, build, and publish the release
* Publish `storybook` by hand.

#### Generate CHANGELOG automatically
Generate CHANGELOG according to two different release content.

#### Publish packages
* Bump packages version separately. 

* [lerna#--skip-npm](https://github.com/lerna/lerna#--skip-npm) for test.

#### Publish storybook
**3** steps for publishing `storybook`. Example: [chun-mingchen.github.io/react-bootstrap-table2/](https://chun-mingchen.github.io/react-bootstrap-table2/)

* clean
* build
* deploy

deploy directory `storybook-static` as site's source files to branch `gh-pages`.

### CODE REVIEW

File: packages/react-bootstrap-table2-example/package.json [hunk 1]
```diff
@@ -3,10 +3,14 @@
   "version": "0.0.1",
   "description": "",
   "main": "index.js",
+  "private": true,
```
Comment (Chun-MingChen) [inline packages/react-bootstrap-table2-example/package.json]: > Lerna won't publish packages which are marked as private ("private": true in the package.json).

ref: https://github.com/lerna/lerna#publish

File: lerna.json [hunk 2]
```diff
@@ -3,5 +3,15 @@
   "packages": [
     "packages/*"
   ],
-  "version": "0.0.0"
+  "version": "independent",
```
Comment (Chun-MingChen) [inline lerna.json]: It will bump packages versions separately.

File: package.json [hunk 3]
```diff
@@ -41,13 +47,16 @@
     "eslint-plugin-import": "2.7.0",
     "eslint-plugin-jsx-a11y": "5.1.1",
     "eslint-plugin-react": "7.2.1",
+    "git-directory-deploy": "1.5.1",
     "html-webpack-plugin": "2.30.1",
     "jest": "20.0.4",
     "jsdom": "11.2.0",
     "jsdom-global": "3.0.2",
     "lerna": "2.0.0",
+    "lerna-changelog": "^0.7.0",
```
Comment (AllenFang) [inline package.json]: No `^`

File: packages/react-bootstrap-table2-example/package.json (hunk 1 above)
Comment (AllenFang) [reply packages/react-bootstrap-table2-example/package.json]: Cool

File: packages/react-bootstrap-table2-example/package.json [hunk 4]
```diff
@@ -3,10 +3,14 @@
...
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
     "storybook": "start-storybook -p 6006",
-    "build-storybook": "build-storybook"
+    "gh-pages:clean": "rimraf ./storybook-static",
+    "gh-pages:build": "build-storybook",
+    "gh-pages:deploy": "git-directory-deploy --directory storybook-static",
```
Comment (AllenFang) [inline packages/react-bootstrap-table2-example/package.json]: Should I install `git-directory-deploy` in my local?

**Comment by AllenFang** [comment]
Go #167

**Pull request closed without merging**